Sources/PackageLoading/IdentityResolver.swift

```diff
@@ -0,0 +1,40 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import TSCBasic
+import PackageModel
+
+// TODO: refactor this when adding registry support
+public protocol IdentityResolver {
+    func resolveIdentity(for location: String) -> PackageIdentity
+    func resolveIdentity(for path: AbsolutePath) -> PackageIdentity
+    func resolveLocation(from location: String) -> String
```

Comment:
Is this `String` because it could be either path or URL or identity?

Reply:
yes, I was contemplating moving to URL as well but wanted to minimize changes so kept String for now. We could/should revisiting this in follow-up PRs

Reply:
Great, that seems like a good choice (keeping PRs incremental)